### PR TITLE
feat: Implement `--annotation-style` parameter for `uv pip compile`

### PR DESCRIPTION
**Pull request by DrJackilD**
## Summary

Hello there! The motivation for this feature is described here #1678 

## Test Plan

I've added unit tests and also tested this manually on my work project by comparing it to the original `pip-compile` output - it looks much like the `pip-compile` generated lock file.

### CODE REVIEW

**Comment by zanieb** [comment]
Thanks for contributing! Welcome to the project :)

I'm not entirely sure we want to accept more configuration here, I'll need to check with @charliermarsh.

It looks like the Windows snapshots are failing. We automatically filter out some Windows-only dependencies, but it's failing since they're not separated by a newline anymore. You'll need to adjust the regex at https://github.com/astral-sh/uv/blob/ad12d97e714aefdf79267faa8c51a36d4783459c/crates/uv/tests/common/mod.rs#L261-L264

**Review by frayandy — APPROVED** [review]

**Comment by DrJackilD** [comment]
@zanieb I hope you'll accept this one :) Regarding the tests - fixed!

File: crates/uv-resolver/src/resolution.rs [hunk 1]
```diff
@@ -376,21 +418,30 @@ impl std::fmt::Display for DisplayResolutionGraph<'_> {
...
+                    AnnotationStyle::Split => {
+                        if !edges.is_empty() {
+                            sep = "\n";
                         }
+                        annotation = DisplayResolutionGraph::annotation_split(edges);
                     }
                 }
             }
+            // Assemble the line with the annotations and remove trailing whitespaces.
+            line = format!("{line:24}{sep}{annotation}")
```
Comment (charliermarsh) [inline crates/uv-resolver/src/resolution.rs:437]: Can you walk me through `line:24` here?

**Comment by charliermarsh** [comment]
I'm happy to include this.

File: crates/uv-resolver/src/resolution.rs (hunk 1 above)
Comment (DrJackilD) [reply crates/uv-resolver/src/resolution.rs:437]: Sure! I'm trying to keep the same logic, as `pip-tools`, to minimize the diff in the output file. The align `24` is something they use in the `pip-tools` as a reasonable number to align the annotation comments: https://github.com/jazzband/pip-tools/blob/main/piptools/writer.py#L318 and I agree with them. Most of the times annotation would be aligned except some extra-long dependencies.

```requirements.txt
monotonic==1.6            # via analytics-python
oauthlib==2.1.0           # via pyproject.toml
opentelemetry-api==1.22.0  # via ddtrace
packaging==21.3           # via gunicorn, pyproject.toml
prompt-toolkit==3.0.43    # via click-repl
protobuf==4.25.3          # via ddsketch, ddtrace
```

Comment (DrJackilD) [reply crates/uv-resolver/src/resolution.rs:437]: Do you mean to extract `annotation_line` and `annotation_split` onto the `match` statement? I could do this, they're not very big, should be fine

Comment (DrJackilD) [reply crates/uv-resolver/src/resolution.rs:428]: Yeah, you're right, it's easier to read without those function calls :)

**Comment by DrJackilD** [comment]
I have no idea, why it's failing now 🤔

**Comment by charliermarsh** [comment]
Will review and merge tonight, hopefully.

**Review by charliermarsh — APPROVED** [review]

**Comment by charliermarsh** [comment]
Thank you! Great contribution.

**Comment by DrJackilD** [comment]
> Thank you! Great contribution.

Thank you for accept it so quick!

**Pull request merged**